server/src/main/java/com/cloud/storage/listener/StoragePoolMonitor.java

```diff
@@ -108,10 +108,12 @@ public void processConnect(Host host, StartupCommand cmd, boolean forRebalance)
                 List<StoragePoolVO> zoneStoragePoolsByAnyHypervisor = _poolDao.findZoneWideStoragePoolsByHypervisor(host.getDataCenterId(), HypervisorType.Any);
                 pools.addAll(zoneStoragePoolsByAnyHypervisor);
 
+                // get the disabled pools list if global setting is true.
+                if (StorageManager.MountDisabledStoragePool.value()) {
+                    pools.addAll(_poolDao.findDisabledPoolsByScope(host.getDataCenterId(), null, null, ScopeType.ZONE));
```

Comment:
@ravening 
how about cluster wide storage pools ?

Reply:
@weizhouapache same comment from me here :) https://github.com/apache/cloudstack/pull/6164/files#r835471732

Reply:
ah, ok thanks @nvazquez 

@ravening any reason why cluster-wide pools are not supported ? can you extend this PR to support cluster-wide pools ?

Reply:
@ravening since VM's can be migrated across clusters, I want to support only zone wide storage pools. If disabled pool is only mounted in cluster A but if we want to migrate vm to cluster B then its not possible.

Reply:
@ravening 
ok. I am not quite clear what you want to fix/achieve. can you explain a bit more ?

anyway, the support of cluster-wide pools is an improvement of this pr. it means not only zone-wide pools , but also cluster-wide pools are mounted. any issue ?

Reply:
@weizhouapache when we set the status of the storage pool as disabled, it will not be used for vm allocation but it will be still mounted/accessible for the vm's. When the node reboots, all the disabled pools will not be mounted and as a result the volume of the vm will not be accessible.

My understanding of the cluster wide pool is that it will be mounted on the hosts within the same cluster right? The same pool will not be mounted on the nodes of different cluster.


Lets say we have three clusters. A, B and C.
I want the disabled pool to be mounted on all three clusters rather than just on A or B or C. Thats why I am searching for zone wide pools

Reply:
thanks @ravening 

if node reboots, you cannot acess the disabled pools (and the volumes on the pools) because they are not mounted on node. is it the issue you want to fix ?

zone-wide pools will be mounted on all hosts in the zone. cluster-wide pools will be mounted on all hosts in specific cluster. I understand you use only zone-wide pools, but many other users use cluster-wide pools. why not mount cluster-wide pools as well ?

Reply:
@weizhouapache  you are saying that, I need to make an extra call like this `pools.addAll(_poolDao.findDisabledPoolsByScope(host.getDataCenterId(), host.getClusterId(), null, ScopeType.CLUSTER));` ?

Reply:
> @weizhouapache you are saying that, I need to make an extra call like this `pools.addAll(_poolDao.findDisabledPoolsByScope(host.getDataCenterId(), host.getClusterId(), null, ScopeType.CLUSTER));` ?

@ravening 
yes exactly. it would be good to have it.
please also consider change it to a cluster level setting ?

Reply:
@weizhouapache @nvazquez added cluster wide support. please review again